File: app/src/main/java/com/adaldosso/spendykt/api/Expense.java
package com.adaldosso.spendykt.api;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import static com.adaldosso.spendykt.utils.SpendyUtils.extractMonth;
import static com.adaldosso.spendykt.utils.SpendyUtils.extractYear;

/**
 * Expense pojo
 *
 * Created by Alberto Dal Dosso on 02/10/2017.
 */

public class Expense extends BaseExpense {

    @SerializedName("date")
    @Expose
    private String date;

    public Expense(String year) {
        super(year);
    }

    public String getYear() {
        return extractYear(date);
    }

    public String getMonth() {
        return extractMonth(date);
    }

    public String getDate() {
        return date;
    }
}

File: app/src/main/java/com/adaldosso/spendykt/api/Expense.kt
package com.adaldosso.spendykt.api

import com.adaldosso.spendykt.utils.SpendyUtils.extractMonth
import com.adaldosso.spendykt.utils.SpendyUtils.extractYear
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Expense pojo
 *
 * Created by Alberto Dal Dosso on 02/10/2017.
 */

class Expense(year: String) : BaseExpense(year) {

    @SerializedName("date")
    @Expose
    val date: String? = null

    override var month: String?
        get() = extractMonth(date)
        set(value) {
            super.month = value
        }

    override fun getYear(): String {
        return extractYear(date)
    }
}
